docs/develop/python/interceptors.mdx

```diff
@@ -0,0 +1,192 @@
+keywords:
+  - interceptors
+tags:
+  - Interceptors
+  - Python SDK
+  - Temporal SDKs
+---
+
+The behavior of the Python SDK can be customized in many useful ways by modifying inbound and outbound calls using Interceptors.
+This is similar to the use of "middleware" in web frameworks such as [Django](https://docs.djangoproject.com/en/5.2/topics/http/middleware/), [Starlette](https://www.starlette.io/middleware/), and [Flask](https://flask.palletsprojects.com/en/stable/lifecycle/#middleware).
+
+The methods you implement on your Interceptor classes can perform arbitrary side effects.
+Interceptors can also perform arbitrary modifications to incoming and outgoing data before it is received by the SDK's "real" implementation.
+
+There are five categories of inbound and outbound calls that you can modify in this way:
+
+#### [Outbound Client calls](https://python.temporal.io/temporalio.client.OutboundInterceptor.html)
+
+- `start_workflow()`
+- `signal_workflow()`
+- `list_workflows()`
+- `update_schedule()`
+
+This is not an exhaustive list; refer to the [Python SDK methods](https://python.temporal.io/temporalio.client.OutboundInterceptor.html) for more.
+
+#### [Inbound Workflow calls](https://python.temporal.io/temporalio.worker.WorkflowInboundInterceptor.html)
+
+- `execute_workflow()` (i.e. handle a Workflow Task that is starting a new Workflow Execution)
+- `handle_query()`
+- `handle_signal()`
+- `handle_update_handler()`
+- `handle_update_validator()`
+
+#### [Outbound Workflow calls](https://python.temporal.io/temporalio.worker.WorkflowOutboundInterceptor.html)
+
+- `start_activity()`
+- `start_child_workflow()`
+- `signal_child_workflow()`
+- `signal_external_workflow()`
+- `start_nexus_operation()`
+- `start_local_activity()`
+
+#### [Inbound Activity calls](https://python.temporal.io/temporalio.worker.ActivityInboundInterceptor.html)
+
+- `execute_activity()`
+  - This is technically the only Inbound Activity call, with the same distinction as `execute_workflow()`.
+
+#### [Outbound Activity calls](https://python.temporal.io/temporalio.worker.ActivityOutboundInterceptor.html)
+
+- `info()`
+- `heartbeat()`
+
+Technically, the first of these categories is a Client call, and the remaining 4 are Worker calls.
+It often happens that your Worker and Client interceptors will share code because they implement closely related logic.
+In the Python SDK, you will typically want to create an interceptor class that inherits from _both_ `client.Interceptor` and `worker.Interceptor`, since their method sets do not overlap.
+
+## Client call Interceptors
+
+To modify outbound Client calls, define a class inheriting from [`client.Interceptor`](https://python.temporal.io/temporalio.client.Interceptor.html), and implement the method `intercept_client()` to return an instance of [`OutboundInterceptor`](https://python.temporal.io/temporalio.client.OutboundInterceptor.html) that implements the subset of outbound Client calls that you wish to modify:
+
```

Comment:
How about we move this sentence:

> This example implements an Interceptor on outbound Client calls that sets a certain key in the outbound `headers` field.
A User ID is context-propagated by being sent in a header field with outbound requests.

to here, so that they understand that the code they're about to look at is an example illustating a specific application of interceptors?